assets/js/atomic/blocks/product-elements/price-v2/block.json

```diff
@@ -0,0 +1,49 @@
+{
+	"$schema": "http://schemas.wp.org/trunk/block.json",
+	"apiVersion": 2,
+	"name": "woocommerce/product-price-v2",
+	"title": "Price",
+	"keywords": "WooCommerce",
+	"category": "woocommerce-product-elements",
+	"textdomain": "woo-gutenberg-products-block",
+	"description": "Display the price of a product, including any discounts.",
+	"supports": {
+		"layout": {
+			"allowSwitching": false,
+			"allowInheriting": false,
+			"default": { "type": "flex" }
+		},
+		"html": false
+
+	},
+	"usesContext": [ "queryId", "postId" ],
+	"providesContext": {
+		"woocommerce/isDescendentOfSingleProductTemplate":
+			"isDescendentOfSingleProductTemplate",
+		"woocommerce/isDescendentOfSingleProductBlock":
+			"isDescendentOfSingleProductBlock",
+		"woocommerce/withSuperScriptStyle": "withSuperScriptStyle"
+	},
+	"attributes": {
+		"isDescendentOfSingleProductTemplate": {
+			"type": "boolean",
+			"default": false
+		},
+		"isDescendentOfSingleProductBlock": {
+			"type": "boolean",
+			"default": false
+		},
+		"productId": {
+			"type": "number",
+			"default": 0
+		},
+		"withSuperScriptStyle": {
+			"type": "boolean",
+			"default": false
+		}
+	},
+	"styles": [
+		{ "name": "default", "label": "Default", "isDefault": true },
+		{ "name": "price-super", "label": "Superscript", "isDefault": false }
```

Comment:
Block Styles are usually utilized explicitly for css styles, however given `withSuperScript` also impacts HTML, the selection also triggers attribute/context update.